### PR TITLE
feat(frontend): [PYTA-38] add theme provider and mode toggle components

### PR DESCRIPTION
**Pull request by mateusz-kurowskiUG**

### CODE REVIEW

File: frontend/src/app/[locale]/_components/ModeToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export function ModeToggle() {
+	const { setTheme } = useTheme();
+
+	return (
+		<DropdownMenu>
+			<DropdownMenuTrigger asChild>
+				<Button variant="outline" size="icon">
+					<Sun className="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0" />
+					<Moon className="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100" />
+					<span className="sr-only">Toggle theme</span>
```
Comment (olszewskista) [inline frontend/src/app/[locale]/_components/ModeToggle.tsx:24]: where internationalization

Comment (mateusz-kurowskiUG) [reply frontend/src/app/[locale]/_components/ModeToggle.tsx:24]: good good

File: frontend/src/app/[locale]/_components/ModeToggle.tsx [hunk 2]
```diff
@@ -11,28 +11,29 @@ import {
...
 					<Sun className="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0" />
 					<Moon className="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100" />
-					<span className="sr-only">Toggle theme</span>
+					<span className="sr-only">t('changeMode')</span>
 				</Button>
 			</DropdownMenuTrigger>
 			<DropdownMenuContent align="end">
 				<DropdownMenuItem onClick={() => setTheme("light")}>
-					Light
+					t('ligth')
```
Comment (olszewskista) [inline frontend/src/app/[locale]/_components/ModeToggle.tsx:30]: gosciu

Comment (mateusz-kurowskiUG) [reply frontend/src/app/[locale]/_components/ModeToggle.tsx:30]: xD

**Review by olszewskista — APPROVED** [review]

**Review by Memetelve — APPROVED** [review]

**Pull request merged**